controllers/machinepools.go

```diff
@@ -34,80 +35,109 @@ func createOCMConnection() (*ocmsdk.Connection, error) {
 	return connection, nil
 }
 
-func scaleMachinePool(aw *arbv1.AppWrapper, userRequestedInstanceType string, replicas int) {
-	connection, err := createOCMConnection()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "Error creating OCM connection: %v", err)
-		return
+func hasAwLabel(machinePool *cmv1.MachinePool, aw *arbv1.AppWrapper) bool {
+	value, ok := machinePool.Labels()[aw.Name]
+	if ok && value == aw.Name {
+		return true
 	}
-	defer connection.Close()
+	return false
+}
 
-	clusterMachinePools := connection.ClustersMgmt().V1().Clusters().Cluster(ocmClusterID).MachinePools()
+func (r *AppWrapperReconciler) scaleMachinePool(ctx context.Context, aw *arbv1.AppWrapper, demandPerInstanceType map[string]int) (ctrl.Result, error) {
+	for userRequestedInstanceType := range demandPerInstanceType {
+		replicas := demandPerInstanceType[userRequestedInstanceType]
+		connection, err := r.createOCMConnection()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error creating OCM connection: %v", err)
+			return ctrl.Result{}, err
+		}
+		defer connection.Close()
 
-	m := make(map[string]string)
-	m[aw.Name] = aw.Name
+		clusterMachinePools := connection.ClustersMgmt().V1().Clusters().Cluster(r.ocmClusterID).MachinePools()
 
-	machinePoolID := strings.ReplaceAll(aw.Name+"-"+userRequestedInstanceType, ".", "-")
-	createMachinePool, err := cmv1.NewMachinePool().ID(machinePoolID).InstanceType(userRequestedInstanceType).Replicas(replicas).Labels(m).Build()
-	if err != nil {
-		klog.Errorf(`Error building MachinePool: %v`, err)
-	}
-	klog.Infof("Built MachinePool with instance type %v and name %v", userRequestedInstanceType, createMachinePool.ID())
-	response, err := clusterMachinePools.Add().Body(createMachinePool).SendContext(context.Background())
-	if err != nil {
-		klog.Errorf(`Error creating MachinePool: %v`, err)
+		response, err := clusterMachinePools.List().SendContext(ctx)
+		if err != nil {
+			klog.Errorf("error retrieving machine pools: %v", err)
+		}
+
+		numberOfMachines := 0
+		response.Items().Each(func(machinePool *cmv1.MachinePool) bool {
+			if machinePool.InstanceType() == userRequestedInstanceType && hasAwLabel(machinePool, aw) {
+				numberOfMachines = machinePool.Replicas()
+				return false
+			}
+			return true
+		})
+
+		if numberOfMachines != replicas {
+			m := make(map[string]string)
+			m[aw.Name] = aw.Name
+			klog.Infof("The instanceRequired array: %v", userRequestedInstanceType)
+
+			machinePoolID := strings.ReplaceAll(aw.Name+"-"+userRequestedInstanceType, ".", "-")
+			createMachinePool, err := cmv1.NewMachinePool().ID(machinePoolID).InstanceType(userRequestedInstanceType).Replicas(replicas).Labels(m).Build()
+			if err != nil {
+				klog.Errorf(`Error building MachinePool: %v`, err)
+			}
```

Comment:
The remainder of the operations after this will return an error if building the machine pool returns an error